runtime/caml/misc.h

```diff
@@ -174,9 +178,15 @@ CAMLdeprecated_typedef(addr, char *);
 /* Prefetching */
 
 #ifdef CAML_INTERNALS
-#if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
+#if (__has_builtin(__builtin_prefetch) || defined(__GNUC__)) && \
```

Comment:
I think that eventually, that conditional should be simply `#if has_builtin(__builtin_prefetch)` so that other platforms (at least arm64, power and s390x with gcc) may benefit from this. But that's out of the scope of that PR.

Reply:
Yes, the original PR authors were conservative in enabling prefetching on other platforms, as they missed data.

> > Just curious: why only on x86? __builtin_prefetch exists on all GCC-supported platforms, even though it can be a no-op. And I'm sure ARM and others would benefit too.
> 
> I left others out because I don't really know anything about non-x86 memory hierarchies. We can turn it on for ARM if you like, but I can't judge how much / whether it'll help, and I don't have the expertise / time to do any serious benchmarking.

https://github.com/ocaml/ocaml/pull/9934#discussion_r499563023